docs/architecture.md

```diff
@@ -41,6 +41,19 @@ Here's a diagram of what that might look like:
 
 <img src={require('@site/static/img/architecture/api-worker.png').default} alt="DBOS Architecture" width="750" className="custom-img"/>
 
+## How DBOS Scales
+
+Fundamentally, DBOS scales with the database it is connected to.
+The only overhead DBOS adds is database writes: one database write per step (to checkpoint the step's outcome) plus two additional database writes per workflow (one at the beginning to checkpoint workflow inputs, one at the end to checkpoint the workflow outcome).
+
+As these writes are checkpointing workflow inputs and outputs and step outputs, the sizes of the writes are determined by the sizes of your inputs and outputs.
+If your steps return small objects, the write sizes will be negligible, but if they return large files, the write sizes will be large.
+Thus, we recommend architecting steps to avoid large output sizes.
```

Comment:
We can be more explicit here: for large files, consider storing the actual data in cheap cloud storage like S3, and return a pointer to that file.